photon-ml/src/main/scala/com/linkedin/photon/ml/util/DefaultIndexMap.scala

```diff
@@ -20,10 +20,16 @@ package com.linkedin.photon.ml.util
   *
   * @author yizhou
   */
-class DefaultIndexMap(@transient val featureNameToIdMap: Map[String, Int]) extends IndexMap {
+class DefaultIndexMap(@transient private val featureNameToIdMap: Map[String, Int]) extends IndexMap {
```

Comment:
This is a backward incompatible change and might break things, for example:
GLMSuiteIntegTest.scala:300: value featureNameToIdMap in class DefaultIndexMap cannot be accessed in com.linkedin.photon.ml.util.DefaultIndexMap.

Why do we want to change featureNameToIdMap to private?